File: shade-android-main/app/src/main/java/com/shade/app/data/remote/dto/MessageDto.kt
package com.shade.app.data.remote.dto

import com.google.gson.annotations.SerializedName

/**
 * Inbox queue item — drained from `GET /api/v1/messages/inbox`.
 * [data] is a Base64-encoded [com.shade.app.proto.WebSocketMessage] (same bytes as a WS binary frame).
 */
data class InboxItemDto(
    @SerializedName("data") val data: String,
)

data class InboxResponse(
    @SerializedName("items") val items: List<InboxItemDto> = emptyList(),
    @SerializedName("count") val count: Int = 0,
    @SerializedName("has_more") val hasMore: Boolean = false,
)

data class ReceiptRequest(
    @SerializedName("message_id") val messageId: String,
    @SerializedName("status")     val status: String, // "READ"
)

data class BatchReceiptRequest(
    @SerializedName("receipts") val receipts: List<ReceiptRequest>,
)

File: shade-backend/internal/dto/message_dto.go
package dto

// ReceiptRequest — Android'den gelen tek bir makbuz (REST fallback)
type ReceiptRequest struct {
	MessageID string `json:"message_id"`
	Status    string `json:"status"` // "READ" veya "DELIVERED"
}

// BatchReceiptRequest — POST /messages/receipts body
type BatchReceiptRequest struct {
	Receipts []ReceiptRequest `json:"receipts"`
}

// InboxItem is a queued WebSocketMessage envelope, base64-encoded protobuf.
// Clients should decode and unmarshal into pb.WebSocketMessage — same as WS binary frames.
type InboxItem struct {
	Data string `json:"data"`
}

type InboxResponse struct {
	Items   []InboxItem `json:"items"`
	Count   int         `json:"count"`
	HasMore bool        `json:"has_more"`
}
